src/webgpu/shader/execution/expression/unary/indirection.spec.ts

```diff
@@ -0,0 +1,198 @@
+export const description = `
+Execution Tests for unary indirection (dereference)
+`;
+
+import { TestParams } from '../../../../../common/framework/fixture.js';
+import { makeTestGroup } from '../../../../../common/framework/test_group.js';
+import { GPUTest } from '../../../../gpu_test.js';
+import { ScalarKind, scalarType } from '../../../../util/conversion.js';
+import { sparseScalarF32Range } from '../../../../util/math.js';
+import {
+  allButConstInputSource,
+  basicExpressionWithPredeclarationBuilder,
+  run,
+  ShaderBuilder,
+} from '../expression.js';
+
+export const g = makeTestGroup(GPUTest);
+
+// All the ways to deref an expression
+const kDerefTypes = [
+  'DerefAddressOfIdentifier', // *(&a)
```

Comment:
nit: most strings that end up in case parameters are `snake_case`

Reply:
Done.